app/src/main/java/org/mozilla/focus/session/SessionNotificationService.kt

```diff
@@ -150,17 +151,24 @@ class SessionNotificationService : Service() {
             val intent = Intent(context, SessionNotificationService::class.java)
             intent.action = ACTION_START
 
-            if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.O) {
-                context.startForegroundService(intent)
-            } else {
-                context.startService(intent)
-            }
+            // Post starting the service to the end of the main thread to give it the best
+            // chance to complete in time.
+            ThreadUtils.postToMainThread(Runnable {
+                if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.O) {
+                    context.startForegroundService(intent)
```

Comment:
Can you check what happens when this is called when the app is in the background? e.g. post to main thread with a delay of ten seconds, open app, press home button, and wait for foreground service notification to start and if it doesn't show up, we have a problem.

I'm concerned this may crash if called when the app is stopped. Possible solution: use LiveData/ViewModel because LiveData doesn't call its observers unless the app is started.

Reply:
@mcomella 

Good idea. After checking the notification service shows up after 10 seconds and it works as intended.